client/src/shared/services/heroFactory.js

```diff
@@ -2,22 +2,67 @@
   'use strict';
 
   angular.module('app')
-  .factory('heroFactory', [function() {
-    var order = {};
+    .factory('heroFactory', [function() {
+      var order = {
+        transactionId: undefined,
+        queueHero: undefined,
+        requester: undefined,
+        item: undefined,
+        moneyExchanged: undefined,
+        vendor: undefined,
+        meetingLocation: undefined,
+        currentLocation: undefined,
+        meetingTime: undefined,
+        status: undefined
+      };
 
-    return {
-      getOrder: getOrder,
-      setOrder: setOrder
-    };
+      return {
+        getOrder: getOrder,
+        setOrder: setOrder
+      };
 
-    function getOrder() {
-      return order;
-    }
+      function getOrder(keys) {
+        if (keys === undefined) {
+          return order;
+        }
 
-    function setOrder(updatedOrder) {
-      order = updatedOrder;
-    }
+        if (angular.isArray(keys)) {
+          var results = {};
+          for (var i = 0; i < keys.length; i++) {
+            if (keys[i] in order) {
+              results[keys[i]] = order[keys[i]];
+            } else {
+              results[keys[i]] = null;
+            }
+          }
+          return results;
+        }
 
-  }]);
+        if (angular.isString(keys)) {
+          if (keys in order) {
+            return order[keys];
+          }
+        }
+

```

Comment:
Sometimes there is a empty line before your return statement, sometimes there isn't. No need to fix now.